src/pages/index.js

```diff
@@ -65,34 +109,137 @@ profileEditBtn.addEventListener("click", () => {
   editModal.open();
 });
 
-//--------------------ADD CARD POPUP-WITH-FORM INSTANTIATION-------------------->>
+//--------------------PROFILE PICTURE POPUP-WITH-FORM INSTANTIATION-------------------->>
+
+const profilePictureModal = new PopupWithForm(
+  "#profile-picture-edit-modal",
+  () => {
+    profilePictureModal.loadingButtonState();
+    api
+      .setProfilePicture(profilePictureFormInput.value)
```

Comment:
`      .setProfilePicture(profilePictureFormInput.value)
`

Only method `_getInputValues` collects all the data of the inputs and passes it to the call of `submitForm`. So you need to use the collected data rather than the avatar input.


It means you get them in line 116


```js

  (values) => {  // here
    profilePictureModal.loadingButtonState();
    api
      .setProfilePicture(values)
```

Reply:
> ` .setProfilePicture(profilePictureFormInput.value)`
> 
> Only method `_getInputValues` collects all the data of the inputs and passes it to the call of `submitForm`. So you need to use the collected data rather than the avatar input.
> 
> It means you get them in line 116
> 
> ```js
>   (values) => {  // here
>     profilePictureModal.loadingButtonState();
>     api
>       .setProfilePicture(values)
> ```

Could you explain what you want me to change here in a little bit more detail? As it stands currently, using values as an argument in my submit handler callback and passing it as an argument to api.setProfilePicture just results in 400 errors.

Reply:
I understand that you want me to use getInputValues to retrieve the response and then pass the collected data, but I am confused where I need to do this.

Reply:
Alright, I think I solved it